### DISCUSSION TITLE
Skipping a submodule in a multimodule maven project

### DISCUSSION THREAD

**Discussion by TomerFi** [sormuras/junit-platform-maven-plugin#74] [Q&A] [answered]
Hey

I stumbled into a weird case.
I can spin up a demo repository, but I figured I'll ask first.

I have the following maven reactor project layout:
- parent (pom)
  - child1 (includes main and test sources)
  - child2 (includes main and test sources)
  - child3 (used for coverage report aggregation and has **no main nor test sources**)
  - child4 (used for benchmarking and has **only main sources and no test sources**)

The `junit-platform-maven-plugin` is configured as an extension at the `parent`'s pom.

While building:
When `child3`, which has no main nor test sources, gets tested, the plugin spits out:

```shell
JUnit Platform Plugin execution skipped: test output directory does not exist.
```

As an *INFO* message and doesn't fail the build.

But when `child4`, which has main sources but no test sources, gets tested, the plugin fails, spitting out:

```shell
Source directory doesn't exists (...\parent\child4\target\test-classes)
```

I've tried playing around with `tweaks.skipOnMissingTestOutputDirectory` but to no avail.

Any thoughts on how can I get `junit-platform-maven-plugin` to skip a specific submodule like `child4`?

Thank you!
:-)

**Comment by sormuras** [comment, marked as answer]
Hm... 🤔 ...it should not fail. Just like with the `child3` submodule. Do you have a stacktrace for me?

**Reply by sormuras** [reply]
According to...

https://github.com/sormuras/junit-platform-maven-plugin/blob/206b83e2d1f13e970c0c38475eefc947c9f75076/src/main/java/de/sormuras/junit/platform/maven/plugin/JUnitPlatformMojo.java#L403-L412

...the `skipOnMissingTestOutputDirectory` tweak (which is enabled by default) is evaluated if no user-specified selector is configured. IIRC, if you configure a custom selector, you are in charge to adapt it (per Maven module) accordingly.

**Reply by TomerFi** [reply]
No selectors selected.
:-)

I was able to create a demo project with minimal configuration representing my case.
You can have a look [here](https://github.com/TomerFi/junit-platform-playground).

I should point out that it happens only when testing in a modular environment.
I have two branches in my repository and a manual dispatched workflow for testing:

- for the [non-modular](https://github.com/TomerFi/junit-platform-playground/tree/non-modular) branch, the [test workflow](https://github.com/TomerFi/junit-platform-playground/runs/2692277823) succeeds.
- for the [modular](https://github.com/TomerFi/junit-platform-playground/tree/modular) branch, the [test workflow](https://github.com/TomerFi/junit-platform-playground/runs/2692288174?check_suite_focus=true) failes.

**Reply by TomerFi** [reply]
As for the stack trace,
I've pasted it [here (pastebin)](https://pastebin.com/GicT66fT).

**Reply by TomerFi** [reply]
Hey @sormuras 

Did you happen to find any leads on this?
For now, as a workaround, I've added a dummy test to child4.

**Reply by sormuras** [reply]
> For now, as a workaround, I've added a dummy test to child4.

Which is an easy way to by-pass this ... bug. ;-)

Didn't have to time to investigate the underlying issue, though.

**Reply by TomerFi** [reply]
Hey,

Bumped into this again today,
and figured I'll ping you.
:-)

**Reply by sormuras** [reply]
_Pong..._

Would you mind porting your playground project into an [integration test](https://github.com/sormuras/junit-platform-maven-plugin/tree/master/src/it) and provide a (failing) PR? Then we can start fixing this annoying ... bug.

**Reply by TomerFi** [reply]
Will do.
:-)

**Reply by TomerFi** [reply]
@sormuras https://github.com/sormuras/junit-platform-maven-plugin/pull/79.
